### ISSUE TITLE
Looking at a record from different collection uses wrong form

### ISSUE THREAD

**Issue by maxxxxxdlp** [specify/specify7#1886]
When looing at a record from a different collection, which has a form that differs from the current collection, the record is displayed using the form from the current collection.

Can be replaced by running query builder and pressing "Browse in forms" on a resulting record.

Another place where this happens:
> If you look at a CO from a relationship in the Voucher collection, the CO record displays using the forms for the Voucher collection rather than the Tissue collection.

Reported by CSIRO

**Comment by maxxxxxdlp** [comment]
Back-end needs to extend the `/context/view.json` api to allow explicitly specifying the collection a view should be resolved for.

**Comment by benanhalt** [comment]
This is already possible by doing `/context/view.json?collectionid=xyz`. But there is also the schema localization to consider if the collection is in a different discipline.

**Comment by benanhalt** [comment]
The way this was handled in previous version of 7 was to switch the user to the correct collection for the resource. Not sure if you have changed that, Max.

**Comment by maxxxxxdlp** [comment]
No, I didn't change that. But that code only checks when opening the record directly.

It does not work when opening a record in a dialog as part of the query builder's "Browse In Forms" feature or the Collection Relationship plugin.

**Comment by maxxxxxdlp** [comment]
Actually, it looks like query results are already scoped to the current collection when running a query on CO:

<img width="1800" alt="Screen Shot 2022-07-27 at 19 58 06" src="https://user-images.githubusercontent.com/40512816/181397669-3ca9f3ef-2a70-46db-b796-d5e7f0bbbb1d.png">

Though, they are not scoped when running a query on Collection table (which makes sense):
<img width="1800" alt="Screen Shot 2022-07-27 at 20 00 48" src="https://user-images.githubusercontent.com/40512816/181397970-027550f5-6d74-40bf-a549-5d526aabcfb9.png">

**Comment by maxxxxxdlp** [comment]
@benanhalt Actually, I am confused by the query builder behavior when restricting results to current collection

For example, running the following query in the voucher collection:
<img width="1800" alt="Screen Shot 2022-07-27 at 20 22 01" src="https://user-images.githubusercontent.com/40512816/181399995-6a758451-0406-47fb-aaec-ebc3ecc7a861.png">

Says that this collection object belongs to the voucher collection:
<img width="1800" alt="Screen Shot 2022-07-27 at 20 22 38" src="https://user-images.githubusercontent.com/40512816/181400072-5f34359f-9240-49d4-b6cb-a1ad36001435.png">

But, running it in the teaching collection says that the CO belongs to the teaching collection:
<img width="1800" alt="Screen Shot 2022-07-27 at 20 22 09" src="https://user-images.githubusercontent.com/40512816/181400013-78320492-96da-492f-b565-5f2d0c0585a2.png">

**Comment by benanhalt** [comment]
These are two different collection objects with the same catalog number. The catalog number is unique within a collection, but not across collections generally speaking.

**Comment by maxxxxxdlp** [comment]
For future reference:
This is much more than just using wrong form.
It is also using from field labels, data obj formatters, UI formatters and permissions.

Sp7 and Sp6 currently assume that the user sees just one collection at a time and changing that assumption would be hard.

A workaround might be to add a way to indicate to specify that even though the user is signed in in one collection, they want to see data from another collection (i.e, though a query string parameter). Then, specify 7 with that query string parameter is going to be embedded inside a dialog box. If specify 7 has that query string parameter, it also needs to somehow pass that argument in all requests for app resources
